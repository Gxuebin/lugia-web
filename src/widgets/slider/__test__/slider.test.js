import * as React from 'react';
import Slider from '../slider';
import Wrapper from '../demo';

import chai from 'chai';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Enzyme, { mount, shallow } from 'enzyme';
import 'jest-styled-components';
const { expect: exp } = chai;
Enzyme.configure({ adapter: new Adapter() });
describe('default', () => {
  it('Wrapper', () => {
    const Target = <Slider />;
    expect(renderer.create(Target).toJSON()).toMatchSnapshot();
  });
  it('defaultValue :8', () => {
    const target = mount(<Slider maxValue={30} defaultValue={8} />);
    expect(target.state().value).toBe(8);
  });
  it('value :2', () => {
    const Target = mount(<Slider maxValue={30} defaultValue={8} value={2} />);
    Target.find('Button')
      .at(0)
      .simulate('mousedown');
    expect(Target.state().value).toBe(2);
  });
  it('disabled', () => {
    const target = mount(<Slider value={2} maxValue={30} disabled />);
    target
      .find('Button')
      .at(0)
      .simulate('mousedown');
    expect(target.state().value).toBe(2);
  });
  it('btnWidth', () => {
    const target = mount(
      <Slider maxValue={30} defaultValue={8} value={2} disabled btnWidth={'14px'} />
    );
    expect(target.state().value).toBe(2);
  });
  it('onchange', async () => {
    const target = mount(<Slider value={2} maxValue={30} />);
    target
      .find('Button')
      .at(0)
      .simulate('mousedown');
    target.instance().setState({ offsetLeft: 50 });
    target.instance().publicmove(180);
    expect(target.state().value).toBe('13.00');
    target.instance().publicmove(255);
    expect(target.state().value).toBe('20.50');
  });
});
