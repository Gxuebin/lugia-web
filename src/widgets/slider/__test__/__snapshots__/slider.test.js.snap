// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default Wrapper 1`] = `
.c0 {
  width: 300px;
  height: 6px;
  background: #e8e8e8;
  margin-left: 50px;
  border-radius: 6px;
  position: relative;
}

.c1 {
  width: 0%;
  height: 6px;
  background: #684fff;
  border-radius: 6px;
  position: absolute;
  left: 0;
  top: 0;
}

.c2 {
  width: 20px;
  height: 20px;
  border-radius: 50%;
  background: #684fff;
  position: absolute;
  left: 0%;
  top: -7px;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.c3 {
  text-align: center;
  position: absolute;
  left: 50%;
  top: -40px;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  -webkit-transform: translateX(-50%);
}

.c4 {
  display: block;
  min-width: 21px;
  height: 27px;
  line-height: 27px;
  padding: 0 3px;
  background: #666;
  color: #fff;
  border-radius: 3px;
}

.c5 {
  display: inline-block;
  vertical-align: top;
  border-top: 6px solid #666;
  border-left: 5px solid transparent;
  border-right: 5px solid transparent;
  border-bottom: 5px solid transparent;
}

<div
  className="doxx"
>
  <div
    className="c0"
    defaultValue={0}
    disabled={false}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    value={0}
  >
    <div
      className="c1"
      defaultValue={0}
      disabled={false}
      value={0}
    />
    <span
      className="c2"
      defaultValue={0}
      disabled={false}
      onMouseDown={[Function]}
      value={0}
    >
      <span
        className="c3"
      >
        <span
          className="c4"
        >
          0
        </span>
        <span
          className="c5"
        />
      </span>
    </span>
  </div>
</div>
`;
